File: TrelloMini.Frontend/src/app/models/activity-log.ts
import { Board } from './board';
import { List } from './list';
import { Card } from './card';
import { User } from './auth';

export enum ActivityType {
  // Board activities
  BoardCreated = 0,
  BoardUpdated = 1,
  BoardDeleted = 2,
  BoardMemberAdded = 3,
  BoardMemberRemoved = 4,
  BoardMemberRoleChanged = 5,

  // List activities
  ListCreated = 6,
  ListUpdated = 7,
  ListDeleted = 8,
  ListMoved = 9,

  // Card activities
  CardCreated = 10,
  CardUpdated = 11,
  CardDeleted = 12,
  CardMoved = 13,
  CardPriorityChanged = 14,
  CardDueDateChanged = 15,
  CardAssigned = 16,
  CardUnassigned = 17,

  // Collaboration activities
  CommentAdded = 18,
  CommentUpdated = 19,
  CommentDeleted = 20,
  AttachmentAdded = 21,
  AttachmentDeleted = 22,

  // User activities
  UserJoined = 23,
  UserLeft = 24
}

export interface ActivityLog {
  id: number;
  activityType: ActivityType;
  description: string;
  data?: string;
  createdAt: Date;
  userId?: number;
  user?: User;
  boardId?: number;
  board?: Board;
  listId?: number;
  list?: List;
  cardId?: number;
  card?: Card;
  ipAddress?: string;
  userAgent?: string;
}

export interface ActivityLogResponse {
  activities: ActivityLog[];
  totalCount: number;
  page: number;
  pageSize: number;
  hasMore: boolean;
}

File: TrelloMini.Backend/TrelloMini.Api/Models/ActivityLog.cs
using System.ComponentModel.DataAnnotations;

namespace TrelloMini.Api.Models
{
    public enum ActivityType
    {
        // Board activities
        BoardCreated,
        BoardUpdated,
        BoardDeleted,
        BoardMemberAdded,
        BoardMemberRemoved,
        BoardMemberRoleChanged,

        // List activities
        ListCreated,
        ListUpdated,
        ListDeleted,
        ListMoved,

        // Card activities
        CardCreated,
        CardUpdated,
        CardDeleted,
        CardMoved,
        CardPriorityChanged,
        CardDueDateChanged,
        CardAssigned,
        CardUnassigned,

        // Collaboration activities
        CommentAdded,
        CommentUpdated,
        CommentDeleted,
        AttachmentAdded,
        AttachmentDeleted,

        // User activities
        UserJoined,
        UserLeft
    }

    public class ActivityLog
    {
        public int Id { get; set; }

        [Required]
        public ActivityType ActivityType { get; set; }

        [Required]
        [StringLength(500)]
        public string Description { get; set; } = string.Empty;

        // JSON data for additional details
        [StringLength(5000)]
        public string? Data { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // User who performed the action
        public int? UserId { get; set; }
        public User? User { get; set; }

        // Related entities
        public int? BoardId { get; set; }
        public Board? Board { get; set; }

        public int? ListId { get; set; }
        public List? List { get; set; }

        public int? CardId { get; set; }
        public Card? Card { get; set; }

        // IP address for security audit
        [StringLength(45)]
        public string? IpAddress { get; set; }

        // User agent for tracking client
        [StringLength(500)]
        public string? UserAgent { get; set; }
    }
}
